builtin/logical/aws/iam_policies.go

```diff
@@ -0,0 +1,131 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+// PolicyDocument represents an IAM policy document
+type PolicyDocument struct {
+	Version    string           `json:"Version"`
+	Statements StatementEntries `json:"Statement"`
+}
+
+// StatementEntries is a slice of statements that make up a PolicyDocument
+type StatementEntries []interface{}
+
+// UnmarshalJSON is defined here for StatementEntries because the Statement
+// portion of an IAM Policy can either be a list or a single element, so if it's
+// a single element this wraps it in a []interface{} so that it's easy to
+// combine with other policy statements:
+// https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_statement.html
+func (se *StatementEntries) UnmarshalJSON(b []byte) error {
+	var out StatementEntries
+
+	var data interface{}
+	if err := json.Unmarshal(b, &data); err != nil {
+		return err
+	}
+
+	switch t := data.(type) {
+	case []interface{}:
+		out = t
+	case interface{}:
+		out = []interface{}{t}
+	default:
+		return fmt.Errorf("Unsupported data type %T for StatementEntries", t)
```

Comment:
```suggestion
		return fmt.Errorf("unsupported data type %T for StatementEntries", t)
```

Reply:
Added in 64f2dd206ed1b68bac70ca7bc7f1cfa6a54c8325